cmake/OpenCVCompilerOptimizations.cmake

```diff
@@ -170,7 +170,29 @@ elseif(" ${CMAKE_CXX_FLAGS} " MATCHES " -march=native | -xHost | /QxHost ")
   set(CPU_BASELINE_DETECT ON)
 endif()
 
+macro(ocv_force_baseline_detect_and_check_dispatch)
+  set(CPU_BASELINE "DETECT" CACHE STRING "${HELP_CPU_BASELINE}")
+
+  if(CPU_BASELINE STREQUAL "NATIVE")
+    # ok
+  elseif(NOT CPU_BASELINE STREQUAL "DETECT")
+    message(WARNING "Platform ${CMAKE_SYSTEM_PROCESSOR} only supports CPU_BASELINE=DETECT|NATIVE, use CMAKE_CXX_FLAGS or cmake toolchain file options to modify compiler options")
+    set(CPU_BASELINE "DETECT")
```

Comment:
I'm still against forcing of `DETECT`. It could be default, but it should not be forced.
We should not block user to disable some OpenCV intrinsics, e.g. for debug/testing purposes. Even if they are "enabled in compiler".

Reply:
I've added support of an empty `CPU_BASELINE`. Currently this configuration is broken due to incorrect macros being used somewhere in DNN inline optimizations: compiler has feature enabled and some macros are enabled too, but some other macros from generated cv_cpu_config.hpp are disabled, thus causing missing symbols.
```
/usr/bin/ld: modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/cpu_kernels/conv_winograd_f63.cpp.o: in function `cv::dnn::runWinograd63(cv::_InputArray const&, cv::_InputArray const&, cv::_OutputArray const&, cv::Ptr<cv::dnn::FastConv> const&, int, float, float, cv::dnn::dnn4_v20240521::ActivationLayer*, bool)::{lambda(cv::Range const&)#1}::operator()(cv::Range const&) const':
/work/opencv/modules/dnn/src/layers/cpu_kernels/conv_winograd_f63.cpp:205: undefined reference to `cv::dnn::winofunc_BtXB_8x8_F32(float const*, int, float*, int, int, int)'
/usr/bin/ld: modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/cpu_kernels/conv_winograd_f63.cpp.o: in function `cv::dnn::runWinograd63(cv::_InputArray const&, cv::_InputArray const&, cv::_OutputArray const&, cv::Ptr<cv::dnn::FastConv> const&, int, float, float, cv::dnn::dnn4_v20240521::ActivationLayer*, bool)::{lambda(cv::Range const&)#2}::operator()(cv::Range const&) const':
/work/opencv/modules/dnn/src/layers/cpu_kernels/conv_winograd_f63.cpp:302: undefined reference to `cv::dnn::winofunc_accum_F32(float const*, float const*, float*, int, int, int, int, int, int)'
/usr/bin/ld: /work/opencv/modules/dnn/src/layers/cpu_kernels/conv_winograd_f63.cpp:368: undefined reference to `cv::dnn::winofunc_AtXA_8x8_F32(float const*, int, float*, int, float*, int, float, float, float, bool)'
collect2: error: ld returned 1 exit status
```
It is not related to this PR and we'll need to resolve this issue separately, my quick fix attempt has failed.

Reply:
User can disable OpenCV intrinsics on affected platforms in two ways:
* set `-DCPU_BASELINE=` - disable all intrinsics
* set `-DCPU_BASELINE_DISABLE=RVV` - disable selected categories

Example:
```
cmake \
-DRISCV_CLANG_BUILD_ROOT=/chains/llvm-main -DRISCV_GCC_INSTALL_ROOT=/chains/riscv-gcc-14.1.0 -DCMAKE_TOOLCHAIN_FILE=/opencv/platforms/linux/riscv64-clang.toolchain.cmake -DBUILD_SHARED_LIBS=OFF -DWITH_OPENCL=OFF \
-DENABLE_RVV=ON \ # choose RVV flags in the toolchain
-DCPU_BASELINE= \ # disable all intrinsics
-DBUILD_opencv_dnn=OFF \ # know issue
-DCMAKE_BUILD_TYPE=Release \
../opencv
```
CMake output:
```
...
--   CPU/HW features:
--     Baseline:
...
--     C++ flags (Release):         -march=rv64gcv --gcc-toolchain=/chains/riscv-gcc-14.1.0 -w   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winconsistent-missing-override -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -Wno-deprecated-enum-enum-conversion -Wno-deprecated-anon-enum-enum-conversion -fdiagnostics-show-option -pthread -Qunused-arguments -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG  -DNDEBUG
...
```
Generated cv_cpu_config.h:
```.h
// OpenCV CPU baseline features

#define CV_CPU_BASELINE_FEATURES 0 \


// OpenCV supported CPU dispatched features



#define CV_CPU_DISPATCH_FEATURES 0 \
```

-------

If we don't change `CPU_BASELINE` (leave it `DETECT`), then OpenCV intrinsics will be enabled, cv_cpu_config.h:
```
// OpenCV CPU baseline features

#define CV_CPU_COMPILE_RVV 1
#define CV_CPU_BASELINE_COMPILE_RVV 1

#define CV_CPU_BASELINE_FEATURES 0 \
    , CV_CPU_RVV \


// OpenCV supported CPU dispatched features



#define CV_CPU_DISPATCH_FEATURES 0 \
```

Reply:
Lets remove `set(CPU_BASELINE "DETECT")`.
Warning message is enough.

Reply:
Done